tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -61,149 +61,105 @@ int main(int argc, char **argv)
+            EXPECT_EQUAL(size, s2n_stuffer_data_available(&result));
 
-            for (int i = 0; i < LENGTH; i++) {
+            for (size_t i = 0; i < s2n_array_len(test_signature_schemes); i++) {
+                uint16_t iana_value = 0;
                 EXPECT_SUCCESS(s2n_stuffer_read_uint16(&result, &iana_value));
                 EXPECT_EQUAL(iana_value, test_signature_schemes[i]->iana_value);
             }
-
             EXPECT_EQUAL(s2n_stuffer_data_available(&result), 0);
         };
 
         /* Test: do not send unsupported signatures */
         {
-            s2n_stuffer_wipe(&result);
-
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+            conn->security_policy_override = &test_security_policy;
             conn->actual_protocol_version = S2N_TLS12;
-            EXPECT_SUCCESS(s2n_send_supported_sig_scheme_list(conn, &result));
 
+            DEFER_CLEANUP(struct s2n_stuffer result = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&result, 0));
+            EXPECT_OK(s2n_signature_algorithms_supported_list_send(conn, &result));
+
+            uint16_t size = 0;
             EXPECT_SUCCESS(s2n_stuffer_read_uint16(&result, &size));
-            EXPECT_EQUAL(size, s2n_supported_sig_scheme_list_size(conn));
+            EXPECT_EQUAL(size, s2n_stuffer_data_available(&result));
 
-            for (int i = 0; i < LENGTH; i++) {
+            for (size_t i = 0; i < s2n_array_len(test_signature_schemes); i++) {
                 if (test_signature_schemes[i] != &s2n_ecdsa_secp384r1_sha384) {
+                    uint16_t iana_value = 0;
                     EXPECT_SUCCESS(s2n_stuffer_read_uint16(&result, &iana_value));
                     EXPECT_EQUAL(iana_value, test_signature_schemes[i]->iana_value);
                 }
             }
-
             EXPECT_EQUAL(s2n_stuffer_data_available(&result), 0);
-        }
+        };
 
         /* Test: written signatures readable */
         {
-            s2n_stuffer_wipe(&result);
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+            conn->security_policy_override = &test_security_policy;
             conn->actual_protocol_version = S2N_TLS13;
 
-            struct s2n_sig_scheme_list signatures;
+            DEFER_CLEANUP(struct s2n_stuffer result = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&result, 0));
+            EXPECT_OK(s2n_signature_algorithms_supported_list_send(conn, &result));
 
-            EXPECT_SUCCESS(s2n_send_supported_sig_scheme_list(conn, &result));
+            struct s2n_sig_scheme_list signatures = { 0 };
             EXPECT_SUCCESS(s2n_recv_supported_sig_scheme_list(&result, &signatures));
```

Comment:
Nit: Should we change the name of the s2n_recv_supported_sig_scheme_list function as well? Now it doesn't match the send side.

Reply:
Here's my planned contents of s2n_signature_algorithms.h after the full refactor:
```
S2N_RESULT s2n_signature_algorithm_recv(struct s2n_connection *conn, struct s2n_stuffer *in);

S2N_RESULT s2n_signature_algorithms_supported_list_process(struct s2n_connection *conn,
        struct s2n_blob *peer_list);
S2N_RESULT s2n_signature_algorithms_supported_list_recv(struct s2n_connection *conn,
        struct s2n_stuffer *in);

S2N_RESULT s2n_signature_algorithms_supported_list_send(struct s2n_connection *conn,
        struct s2n_stuffer *out);
```

Reply:
Ah I jumped the gun.